Lib/test/test_mailcap.py

```diff
@@ -127,14 +127,18 @@ def test_subst(self):
             (["", "audio/*", "foo.txt"], ""),
             (["echo foo", "audio/*", "foo.txt"], "echo foo"),
             (["echo %s", "audio/*", "foo.txt"], "echo foo.txt"),
-            (["echo %t", "audio/*", "foo.txt"], None),
+            (["echo %t", "audio/*", "foo.txt"], None, {"warn_type": mailcap.UnsafeMailcapInput}),
             (["echo %t", "audio/wav", "foo.txt"], "echo audio/wav"),
             (["echo \\%t", "audio/*", "foo.txt"], "echo %t"),
             (["echo foo", "audio/*", "foo.txt", plist], "echo foo"),
             (["echo %{total}", "audio/*", "foo.txt", plist], "echo 3")
         ]
         for tc in test_cases:
-            self.assertEqual(mailcap.subst(*tc[0]), tc[1])
+            if len(tc) == 3:
+                with warnings_helper.check_warnings(('', tc[2]["warn_type"]), quiet=True):
+                    self.assertEqual(mailcap.subst(*tc[0]), tc[1])
+            else:
+                self.assertEqual(mailcap.subst(*tc[0]), tc[1])
```

Comment:
This seems a bit over-engineered. I'd split this into a separate test function, which tests that the warning is issued in cases that deserve the warning. This exception is raised from 3 places in the Lib/mailcap.py, so shouldn't we have at least 3 test cases for it?

CC @encukou .

Reply:
Hi Irit,

Thank you, I'll rework the affected cases into a separate test function and update the patch.
Will add 2x new test cases for unsafe filename and parameters, only the MIME type warnings are raised at present